include/DoocsAdapter.h

```diff
@@ -2,11 +2,25 @@
 #define _CHIMERATK_DOOCS_ADAPTER_H_
 
 #include "CSAdapterEqFct.h"
-#include "DoocsAdapter.h"
 #include "DoocsPVFactory.h"
 
 namespace ChimeraTK {
 
+  /**
+   * Base class used for all properties.
+   */
+  class PropertyBase : public boost::enable_shared_from_this<PropertyBase> {
+   public:
+    // List of other properties which need to update their DOOCS buffers when this property is written from the DOOCS
+    // side. This is used to synchronise multi-mapped PVs.
+    std::set<boost::shared_ptr<PropertyBase>> otherPropertiesToUpdate;
+
+    // Update DOOCS buffer from PVs. The given transferElementId shall be used only for checking consistency with the
```

Comment:
I am not sure if two concepts are mixed here. The DataConsistencyGroup is for matching the macro pulse number. This has noting to do with other parameters.
The other parameters update must also work if no macro pulse number is defined.

Reply:
Well, this is a base class which needs to reflect several concepts. `updateDoocsBuffer()` needs to become a virtual function which can be called when having merely a pointer to the PropertyBase type. Previously there was only one reason to call this function (corresponding PV changed), now there are two reasons (updates might come instead from another property). Both use cases shall be covered by the same function.

I don't see any problem here?